lib/internal/modules/cjs/loader.js

```diff
@@ -1004,18 +1004,26 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     paths = Module._resolveLookupPaths(request, parent);
   }
 
-  if (parent && parent.filename) {
-    const filename = trySelf(parent.filename, isMain, request);
-    if (filename) {
-      const cacheKey = request + '\x00' +
-          (paths.length === 1 ? paths[0] : paths.join('\x00'));
-      Module._pathCache[cacheKey] = filename;
-      return filename;
-    }
+  let base, filename;
+  if (parent && (parent.filename || parent.id === '<repl>')) {
```

Comment:
This is starting to look complex for logic embedded in this function. It may be cleaner to push this into `trySelf`? There's also two branches (`id == repl` and `id == preload`) that are seemingly doing the same thing or could do the same if we accept the `try` block for the `<repl>` case as well.

Reply:
> This is starting to look complex for logic embedded in this function. It may be cleaner to push this into trySelf?

I'll try that :+1: .

> There's also two branches (id == repl and id == preload) that are seemingly doing the same thing or could do the same if we accept the try block for the <repl> case as well

I think it makes sense, do you agree @guybedford ?

Reply:
Perhaps just separate out the `parent.filename` case from the `<repl>` one then so that the `process.cwd()` checks are unified.